mantidproject/mantid
# Add search option for adding fit functions

**Description of work.**
This PR adds functionality to search for functions in the add function dialog.

**To test:**
Load a workspace and do a plot.
Click the fit button
Right-click on the plot and click any of the `add ...` options.
You should be able to type in the box and the options should appear beneath.
You should also be able to click in the drop down as before. 
Click `ok` and the function should be added as expected.

Right-click on the property browser and click `add function`.
You should be able to search in the box at the top.
Select a function and press `ok` 
The function should be added as expected and you should be able to do a fit.

Fixes #27587 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
